jax/_src/numpy/lax_numpy.py

```diff
@@ -3509,8 +3509,25 @@ def fromstring(string: str, dtype: DTypeLike = float, count: int = -1, *, sep: s
   return asarray(np.fromstring(string=string, dtype=dtype, count=count, sep=sep))
 
 
-@util.implements(np.eye)
+@util.implements(np.eye, extra_params="""
+device : :py:class:`Device`, :py:class:`Sharding`, optional
+  The (optional) :py:class:`Device`, :py:class:`Sharding`,
+  representing the device(s) to which created array should be
+  transferred. If given, then the result is committed to the device(s).
+""")
 def eye(N: DimSize, M: DimSize | None = None,
+        k: int | ArrayLike = 0,
+        dtype: DTypeLike | None = None,
+        device: xc.Device | Sharding | None = None) -> Array:
```

Comment:
Let's make the `device` argument keyword-only

Reply:
@jakevdp can we add `/, *` similarly to what array-api suggests : https://data-apis.org/array-api/latest/API_specification/generated/array_api.eye.html#eye 
```
def eye(N: DimSize, M: DimSize | None = None,
        /, *,
        k: int | ArrayLike = 0,
        dtype: DTypeLike | None = None,
        device: xc.Device | Sharding | None = None) -> Array:
```
?

Reply:
No, we can't change signatures for existing arguments without a deprecation cycle.